src/main/java/uk/gov/digital/ho/hocs/queue/ComplaintService.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.digital.ho.hocs.queue;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.client.ComplaintData;
+import uk.gov.digital.ho.hocs.client.casework.CaseworkClient;
+import uk.gov.digital.ho.hocs.client.workflow.WorkflowClient;
+import uk.gov.digital.ho.hocs.client.workflow.dto.CreateCaseRequest;
+import uk.gov.digital.ho.hocs.client.workflow.dto.CreateCaseResponse;
+
+import java.util.Map;
+import java.util.UUID;
+
+@Slf4j
+@Service
+public class ComplaintService {
+
+    public static final String CORRESPONDENTS_LABEL = "Correspondents";
+    public static final String COMPLAINT_TYPE_LABEL = "ComplaintType";
+    private final WorkflowClient workflowClient;
+    private final CaseworkClient caseworkClient;
+    private final String user;
+
+    @Autowired
+    public ComplaintService(WorkflowClient workflowClient,
+                            CaseworkClient caseworkClient,
+                            @Value("${hocs.user}") String user) {
+        this.workflowClient = workflowClient;
+        this.caseworkClient = caseworkClient;
+        this.user = user;
+    }
+
+    public void createComplaint(ComplaintData complaintData, String caseType) {
+
+        log.info("createComplaint, started");
+
+        CreateCaseRequest request = new CreateCaseRequest(caseType, complaintData.getDateReceived());
+        CreateCaseResponse createCaseResponse = workflowClient.createCase(request);
+
+        UUID caseUUID = createCaseResponse.getUuid();
+
+        log.info("createComplaint, create case : caseUUID : {}", caseUUID);
+
+        UUID stageForCaseUUID = caseworkClient.getStageForCase(caseUUID);
+
+        log.info("createComplaint, get stage for case : caseUUID : {}, stageForCaseUUID : {}", caseUUID, stageForCaseUUID);
+
+        caseworkClient.updateStageUser(caseUUID, stageForCaseUUID, UUID.fromString(user));
+
+        caseworkClient.addCorrespondentToCase(caseUUID, stageForCaseUUID, complaintData.getUkviComplaintCorrespondent());
+
+        UUID primaryCorrespondent = caseworkClient.getPrimaryCorrespondent(caseUUID);
```

Comment:
Is the primary correspondent the same as the correspondent we just added? Do we need to make the extra request here or could we use the info we already have?

Reply:
When we do the add we don't know the correspondent UUID, so we need to make the call to get the UUID so that we can move the case to the next stage. It would have been good if the add returned the UUID, but I'm trying to keep change a minimum to casework. I'm actually copying what the UI does.